src/uast.h

```diff
@@ -0,0 +1,75 @@
+#ifndef _NODE_API
+#define _NODE_API
+
+#include <libxml/parser.h>
+#include <libxml/tree.h>
+#include <libxml/xpath.h>
+#include <libxml/xpathInternals.h>
+
+#include "node-iface.h"
+
+typedef struct _find_ctx {
```

Comment:
I'd propose to only declare the type here, define it in the `.c`, making it private, and to provide some public functions to access it, like:

```
int find_results(find_ctx *ctx);
result* find_result(find_ctx *ctx, int i);
```

The user should know nothing about this internal structure, its `cap` field, etc. That should be encapsulated I'd say.